horiz_interp/include/horiz_interp_conserve.inc

```diff
@@ -468,67 +364,70 @@ contains
     end do
 
     do i = 1, nxgrid
-       Interp%area_frac_dst(i) = xgrid_area(i)/dst_area(Interp%i_dst(i), Interp%j_dst(i) )
+       Interp%HI_KIND_TYPE_%area_frac_dst(i) = xgrid_area(i)/dst_area(Interp%i_dst(i), Interp%j_dst(i) )
     end do
     Interp%nlon_src = nlon_in;  Interp%nlat_src = nlat_in
     Interp%nlon_dst = nlon_out; Interp%nlat_dst = nlat_out
     if(present(mask_out)) then
        if( (size(mask_out,1) .NE. nlon_out) .OR. (size(mask_out,2) .NE. nlat_out) ) call mpp_error(FATAL, &
          'horiz_interp_conserve_mod: size mismatch between mask_out and lon_out/lat_out')
-       mask_out = 0.0
+       mask_out = 0.0_kindl
        do i = 1, nxgrid
           mask_out(Interp%i_dst(i),Interp%j_dst(i)) = mask_out(Interp%i_dst(i), &
-                  & Interp%j_dst(i)) + Interp%area_frac_dst(i)
+                  & Interp%j_dst(i)) + Interp%HI_KIND_TYPE_%area_frac_dst(i)
        end do
     end if
 
     deallocate(i_src, j_src, i_dst, j_dst, xgrid_area, dst_area )
 
-  end subroutine horiz_interp_conserve_new_1dx2d
+  end subroutine HORIZ_INTERP_CONSERVE_NEW_1DX2D_
 
   !#######################################################################
 
-  subroutine horiz_interp_conserve_new_2dx1d ( Interp, lon_in, lat_in, lon_out, lat_out, &
+  subroutine HORIZ_INTERP_CONSERVE_NEW_2DX1D_ ( Interp, lon_in, lat_in, lon_out, lat_out, &
                                                mask_in, mask_out, verbose)
     type(horiz_interp_type),        intent(inout) :: Interp
-    real, intent(in),              dimension(:,:) :: lon_in , lat_in
-    real, intent(in),              dimension(:)   :: lon_out, lat_out
-    real, intent(in),    optional, dimension(:,:) :: mask_in
-    real, intent(inout), optional, dimension(:,:) :: mask_out
+    real(FMS_HI_KIND_), intent(in),              dimension(:,:) :: lon_in , lat_in
+    real(FMS_HI_KIND_), intent(in),              dimension(:)   :: lon_out, lat_out
+    real(FMS_HI_KIND_), intent(in),    optional, dimension(:,:) :: mask_in
+    real(FMS_HI_KIND_), intent(inout), optional, dimension(:,:) :: mask_out
     integer, intent(in), optional                 :: verbose
 
     integer :: create_xgrid_2DX1D_order1, get_maxxgrid, maxxgrid
     integer :: create_xgrid_great_circle
     integer :: nlon_in, nlat_in, nlon_out, nlat_out, nxgrid, i, j
     integer, allocatable, dimension(:)   :: i_src, j_src, i_dst, j_dst
-    real,    allocatable, dimension(:,:) :: dst_area
-    real(DOUBLE_KIND), dimension(size(lon_in,1)-1, size(lon_in,2)-1) :: mask_src
-    real(DOUBLE_KIND), allocatable, dimension(:)   :: xgrid_area, clon, clat
-    real(DOUBLE_KIND), allocatable, dimension(:)   :: lon_out_r8, lat_out_r8
-    real(DOUBLE_KIND), allocatable, dimension(:,:) :: lon_in_r8, lat_in_r8
-    real(DOUBLE_KIND), allocatable, dimension(:,:) :: lon_dst, lat_dst
+    real(FMS_HI_KIND_),    allocatable, dimension(:,:) :: dst_area
```

Comment:
I left it as the macro kind because it doesn't get passed into any xgrid routines, on ln 483 it sets a field in the type to that value  so this way they match up.

Reply:
I think it involved in math with xgrid_area (r8_kind), so perhaps dst_area should be r8_kind too